libraries/chain/webassembly/crypto.cpp

```diff
@@ -255,51 +255,51 @@ namespace eosio { namespace chain { namespace webassembly {
    {
       if(op1.size() != 144 ||  op2.size() != 144 ||  result.size() != 144)
          return return_code::failure;
-      std::optional<bls12_381::g1> a = bls12_381::g1::fromJacobianBytesLE({reinterpret_cast<const uint8_t*>(op1.data()), 144}, false, true);
-      std::optional<bls12_381::g1> b = bls12_381::g1::fromJacobianBytesLE({reinterpret_cast<const uint8_t*>(op2.data()), 144}, false, true);
+      std::optional<bls12_381::g1> a = bls12_381::g1::fromJacobianBytesLE(std::span<uint8_t, 144>((uint8_t*)op1.data(), 144), false, true);
+      std::optional<bls12_381::g1> b = bls12_381::g1::fromJacobianBytesLE(std::span<uint8_t, 144>((uint8_t*)op2.data(), 144), false, true);
       if(!a.has_value() || !b.has_value())
          return return_code::failure;
       bls12_381::g1 c = a.value().add(b.value());
-      c.toJacobianBytesLE({reinterpret_cast<uint8_t*>(result.data()), 144}, true);
+      c.toJacobianBytesLE(std::span<uint8_t, 144>((uint8_t*)result.data(), 144), true);
       return return_code::success;
    }
 
    int32_t interface::bls_g2_add(span<const char> op1, span<const char> op2, span<char> result) const
    {
       if(op1.size() != 288 ||  op2.size() != 288 ||  result.size() != 288)
          return return_code::failure;
-      std::optional<bls12_381::g2> a = bls12_381::g2::fromJacobianBytesLE({reinterpret_cast<const uint8_t*>(op1.data()), 288}, false, true);
-      std::optional<bls12_381::g2> b = bls12_381::g2::fromJacobianBytesLE({reinterpret_cast<const uint8_t*>(op2.data()), 288}, false, true);
+      std::optional<bls12_381::g2> a = bls12_381::g2::fromJacobianBytesLE(std::span<uint8_t, 288>((uint8_t*)op1.data(), 288), false, true);
+      std::optional<bls12_381::g2> b = bls12_381::g2::fromJacobianBytesLE(std::span<uint8_t, 288>((uint8_t*)op2.data(), 288), false, true);
       if(!a.has_value() || !b.has_value())
          return return_code::failure;
       bls12_381::g2 c = a.value().add(b.value());
-      c.toJacobianBytesLE({reinterpret_cast<uint8_t*>(result.data()), 288}, true);
+      c.toJacobianBytesLE(std::span<uint8_t, 288>((uint8_t*)result.data(), 288), true);
       return return_code::success;
    }
 
    int32_t interface::bls_g1_mul(span<const char> point, span<const char> scalar, span<char> result) const
    {
       if(point.size() != 144 ||  scalar.size() != 32 ||  result.size() != 144)
          return return_code::failure;
-      std::optional<bls12_381::g1> a = bls12_381::g1::fromJacobianBytesLE({reinterpret_cast<const uint8_t*>(point.data()), 144}, false, true);
+      std::optional<bls12_381::g1> a = bls12_381::g1::fromJacobianBytesLE(std::span<uint8_t, 144>((uint8_t*)point.data(), 144), false, true);
```

Comment:
Maybe not lose the `const` and pass `std::span<const uint8_t, 144>`. Same in other locations.

Reply:
yep good call